pkg/cli/admin/inspect/inspect.go

```diff
@@ -176,6 +177,14 @@ func (o *InspectOptions) Complete(args []string) error {
 	if err != nil {
 		return err
 	}
+	// TODO (soltysh) change this to set WithManagedFields in PrintFlags creation
+	// above when https://github.com/kubernetes/kubernetes/pull/107947 merges
+	if tsp, ok := printer.(*printers.TypeSetterPrinter); ok {
+		printer = tsp.Delegate
```

Comment:
Seems like we only want to overwrite `printer` on line 185 after we've removed the `OmitManagedFieldsPrinter`.  This line as-is could result in us losing the typesetting capability which is needed for proper serialization.

Reply:
I'm setting back the typer below, just like what we get from invoking `WithTypeSetter` in https://github.com/openshift/oc/blob/3aedcfc3ba5afaa0b7993c3e1824b179a9c241a3/pkg/cli/admin/inspect/inspect.go#L93

Reply:
could we do

```go
if tsp, ok := printer.(*printers.TypeSetterPrinter); ok {
		if omp, ok := tsp.Delegate.(*printers.OmitManagedFieldsPrinter); ok {
			printer = &printers.TypeSetterPrinter{Typer: scheme.Scheme, Delegate: omp.Delegate}
		}
	}
```

to avoid writing `printer` early?